src/backoffice/backoffice.element.ts

```diff
@@ -83,45 +57,20 @@ export class UmbBackofficeElement extends UmbLitElement {
 	constructor() {
 		super();
 
+		// TODO: find a way this is possible outside this element.
 		this.provideContext(UMB_MODAL_CONTEXT_TOKEN, new UmbModalContext());
 		this.provideContext(UMB_NOTIFICATION_CONTEXT_TOKEN, new UmbNotificationContext());
-
-		// TODO: find a way this is possible outside this element. It needs to be possible to register stores in extensions
 		this.provideContext(UMB_CURRENT_USER_STORE_CONTEXT_TOKEN, new UmbCurrentUserStore());
-
-		new UmbDocumentStore(this);
-		new UmbDocumentTreeStore(this);
-		new UmbMediaDetailStore(this);
-		new UmbMediaTreeStore(this);
-		new UmbDataTypeStore(this);
-		new UmbDataTypeTreeStore(this);
-		new UmbUserStore(this);
-		new UmbMediaTypeDetailStore(this);
-		new UmbMediaTypeTreeStore(this);
-		new UmbDocumentTypeStore(this);
-		new UmbDocumentTypeTreeStore(this);
-		new UmbMemberTypeDetailStore(this);
-		new UmbMemberTypeTreeStore(this);
-		new UmbUserGroupStore(this);
-		new UmbMemberGroupDetailStore(this);
-		new UmbMemberGroupTreeStore(this);
-		new UmbMemberDetailStore(this);
-		new UmbMemberTreeStore(this);
-		new UmbDictionaryDetailStore(this);
-		new UmbDictionaryTreeStore(this);
-		new UmbDocumentBlueprintDetailStore(this);
-		new UmbDocumentBlueprintTreeStore(this);
-		new UmbTemplateTreeStore(this);
-		new UmbTemplateDetailStore(this);
-		new UmbLanguageStore(this);
-
 		this.provideContext(UMB_APP_LANGUAGE_CONTEXT_TOKEN, new UmbAppLanguageContext(this));
 		this.provideContext(UMB_BACKOFFICE_CONTEXT_TOKEN, new UmbBackofficeContext());
-		this.provideContext(UMB_CURRENT_USER_HISTORY_STORE_CONTEXT_TOKEN, new UmbCurrentUserHistoryStore());
 		new UmbThemeContext(this);
-
-		new UmbPackageStore(this);
 		new UmbServerExtensionController(this, umbExtensionsRegistry);
+		this.provideContext(UMB_CURRENT_USER_HISTORY_STORE_CONTEXT_TOKEN, new UmbCurrentUserHistoryStore());
+
+		// Register All Stores
+		this.observe(umbExtensionsRegistry.extensionsOfTypes(['store', 'treeStore']), (stores) => {
```

Comment:
Should we add a mechanism for "scope" of store, so you decide if it's registered globally, or do you just not add a manifest for a store, if you want to spin it up elsewhere?

Reply:
I think it is ok to assume a store is always globally available. At least I don't think it is something we need to worry about right now.

Reply:
Sounds good to me.